src/inmanta/agent/cache.py:0: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/inmanta/agent/forking_executor.py:0: error: Argument 1 of "connection_made" is incompatible with supertype "asyncio.protocols.BaseProtocol"; supertype defines the argument type as "BaseTransport"  [override]
src/inmanta/agent/forking_executor.py:0: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
src/inmanta/agent/forking_executor.py:0: note: This violates the Liskov substitution principle
src/inmanta/agent/handler.py:0: error: Argument 3 to "log" of "LogLine" has incompatible type "**dict[str, object]"; expected "datetime | None"  [arg-type]
src/inmanta/agent/handler.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/agent/reporting.py:0: error: Incompatible types in assignment (expression has type "None", variable has type Module)  [assignment]
src/inmanta/agent/resourcepool.py:0: error: Argument 1 to "append" of "list" has incompatible type "Callable[[TPoolMember], Coroutine[Any, Any, bool]]"; expected "Callable[[PoolMember[TPoolID]], Coroutine[Any, Any, Any]]"  [arg-type]
src/inmanta/agent/resourcepool.py:0: error: Missing type parameters for generic type "PoolMember"  [type-arg]
src/inmanta/agent/resourcepool.py:0: error: Missing type parameters for generic type "PoolMember"  [type-arg]
src/inmanta/app.py:0: error: "object" not callable  [operator]
src/inmanta/app.py:0: error: Argument "aliases" to "add_parser" of "_SubParsersAction" has incompatible type "object"; expected "Sequence[str]"  [arg-type]
src/inmanta/app.py:0: error: Argument "help" to "add_parser" of "_SubParsersAction" has incompatible type "object"; expected "str | None"  [arg-type]
src/inmanta/app.py:0: error: Argument 1 to "apply_options" of "InmantaLoggerConfig" has incompatible type "Namespace"; expected "Options"  [arg-type]
src/inmanta/app.py:0: error: Argument 1 to "apply_options" of "InmantaLoggerConfig" has incompatible type "Namespace"; expected "Options"  [arg-type]
src/inmanta/app.py:0: error: Argument 1 to "run" of "Exporter" has incompatible type "dict[str, Type]"; expected "dict[str, Entity] | None"  [arg-type]
src/inmanta/app.py:0: error: Argument 2 to "safe_shutdown" has incompatible type "Callable[[int | None], Coroutine[Any, Any, None]]"; expected "Callable[[], None]"  [arg-type]
src/inmanta/app.py:0: error: Argument 2 to "safe_shutdown" has incompatible type "Callable[[int | None], Coroutine[Any, Any, None]]"; expected "Callable[[], None]"  [arg-type]
src/inmanta/app.py:0: error: Incompatible types in assignment (expression has type "str | None", variable has type "int | None")  [assignment]
src/inmanta/app.py:0: error: Missing type parameters for generic type "Future"  [type-arg]
src/inmanta/app.py:0: error: No overload variant of "print" matches argument types "str", "TextIO | Any", "dict[str, object]"  [call-overload]
src/inmanta/app.py:0: note: Possible overload variants:
src/inmanta/app.py:0: note: def print(*values: object, sep: str | None = ..., end: str | None = ..., file: SupportsWrite[str] | None = ..., flush: Literal[False] = ...) -> None
src/inmanta/app.py:0: note: def print(*values: object, sep: str | None = ..., end: str | None = ..., file: _SupportsWriteAndFlush[str] | None = ..., flush: bool) -> None
src/inmanta/ast/__init__.py:0: error: Argument 1 to "set_location" of "CompilerException" has incompatible type "Location | None"; expected "Location"  [arg-type]
src/inmanta/ast/__init__.py:0: error: Argument 1 to "set_location" of "CompilerException" has incompatible type "Location | None"; expected "Location"  [arg-type]
src/inmanta/ast/__init__.py:0: error: Argument 2 to "__init__" of "RuntimeException" has incompatible type "LocatableString"; expected "Locatable | None"  [arg-type]
src/inmanta/ast/__init__.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/ast/__init__.py:0: error: Missing type parameters for generic type "DelayedResultVariable"  [type-arg]
src/inmanta/ast/__init__.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/__init__.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/__init__.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/__init__.py:0: error: Returning Any from function declared to return "Location"  [no-any-return]
src/inmanta/ast/attribute.py:0: error: Incompatible types in assignment (expression has type "int | None", variable has type "int")  [assignment]
src/inmanta/ast/attribute.py:0: error: Incompatible types in assignment (expression has type "property", variable has type "Type")  [assignment]
src/inmanta/ast/attribute.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/attribute.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/attribute.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/attribute.py:0: error: Need type annotation for "source_annotations" (hint: "source_annotations: list[<type>] = ...")  [var-annotated]
src/inmanta/ast/attribute.py:0: error: Need type annotation for "target_annotations" (hint: "target_annotations: list[<type>] = ...")  [var-annotated]
src/inmanta/ast/blocks.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "ExecutionContext")  [assignment]
src/inmanta/ast/blocks.py:0: error: Name "List" is not defined  [name-defined]
src/inmanta/ast/blocks.py:0: error: Name "List" is not defined  [name-defined]
src/inmanta/ast/blocks.py:0: error: Name "List" is not defined  [name-defined]
src/inmanta/ast/blocks.py:0: error: Returning Any from function declared to return "list[DynamicStatement]"  [no-any-return]
src/inmanta/ast/blocks.py:0: note: Did you forget to import it from "typing"? (Suggestion: "from typing import List")
src/inmanta/ast/blocks.py:0: note: Did you forget to import it from "typing"? (Suggestion: "from typing import List")
src/inmanta/ast/blocks.py:0: note: Did you forget to import it from "typing"? (Suggestion: "from typing import List")
src/inmanta/ast/constraint/expression.py:0: error: Argument 1 to "create_function" has incompatible type "Operator"; expected "ConstraintType"  [arg-type]
src/inmanta/ast/constraint/expression.py:0: error: Argument 2 to "register_operator" of "Operator" has incompatible type "OpMetaClass"; expected "type[Operator]"  [arg-type]
src/inmanta/ast/constraint/expression.py:0: error: Call to untyped function "_op" in typed context  [no-untyped-call]
src/inmanta/ast/constraint/expression.py:0: error: Call to untyped function "_op" in typed context  [no-untyped-call]
src/inmanta/ast/constraint/expression.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/ast/constraint/expression.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/ast/constraint/expression.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/ast/constraint/expression.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/ast/constraint/expression.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/ast/constraint/expression.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/ast/constraint/expression.py:0: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/inmanta/ast/constraint/expression.py:0: error: Missing positional argument "expression" in call to "create_function"  [call-arg]
src/inmanta/ast/constraint/expression.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/constraint/expression.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/constraint/expression.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/constraint/expression.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/constraint/expression.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/constraint/expression.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/constraint/expression.py:0: error: Need type annotation for "temp"  [var-annotated]
src/inmanta/ast/constraint/expression.py:0: error: Returning Any from function declared to return "str"  [no-any-return]
src/inmanta/ast/entity.py:0: error: "type[DataclassInstance]" has no attribute "__dataclass_params__"  [attr-defined]
src/inmanta/ast/entity.py:0: error: Argument "reference" to "AttributeReference" has incompatible type "object"; expected "DataclassInstance | Reference[DataclassInstance]"  [arg-type]
src/inmanta/ast/entity.py:0: error: Argument 1 to "DuplicateException" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/ast/entity.py:0: error: Argument 1 to "fields" has incompatible type "object"; expected "DataclassInstance | type[DataclassInstance]"  [arg-type]
src/inmanta/ast/entity.py:0: error: Argument 2 to "DataClassMismatchException" has incompatible type "type[DataclassInstance]"; expected "DataclassInstance | None"  [arg-type]
src/inmanta/ast/entity.py:0: error: Argument 2 to "DuplicateException" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/ast/entity.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "ExpressionStatement")  [assignment]
src/inmanta/ast/entity.py:0: error: Incompatible types in assignment (expression has type "property", variable has type "dict[str, Attribute]")  [assignment]
src/inmanta/ast/entity.py:0: error: Item "None" of "Attribute | None" has no attribute "type"  [union-attr]
src/inmanta/ast/entity.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/entity.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/entity.py:0: error: Module "inmanta.execute.runtime" does not explicitly export attribute "dataflow"  [attr-defined]
src/inmanta/ast/entity.py:0: error: Returning Any from function declared to return "Location"  [no-any-return]
src/inmanta/ast/entity.py:0: error: Returning Any from function declared to return "Location"  [no-any-return]
src/inmanta/ast/entity.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/ast/entity.py:0: note: ClassVar protocol member DataclassInstance.__dataclass_fields__ can never be matched by a class object
src/inmanta/ast/statements/__init__.py:0: error: Cannot find implementation or library stub for module named "inmanta.ast.assign"  [import-not-found]
src/inmanta/ast/statements/__init__.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/ast/statements/__init__.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "NamedType")  [assignment]
src/inmanta/ast/statements/__init__.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "Namespace")  [assignment]
src/inmanta/ast/statements/__init__.py:0: error: Incompatible types in assignment (expression has type "object", variable has type "ResultCollector[Any] | None")  [assignment]
src/inmanta/ast/statements/__init__.py:0: error: Incompatible types in assignment (expression has type "object", variable has type "Sequence[EagerPromise]")  [assignment]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "ResultCollector"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/__init__.py:0: error: Need type annotation for "target"  [var-annotated]
src/inmanta/ast/statements/__init__.py:0: error: Returning Any from function declared to return "Location"  [no-any-return]
src/inmanta/ast/statements/__init__.py:0: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/inmanta/ast/statements/__init__.py:0: note: Use "-> None" if function does not return a value
src/inmanta/ast/statements/assign.py:0: error: "Type" has no attribute "lookup_index"  [attr-defined]
src/inmanta/ast/statements/assign.py:0: error: "Type" has no attribute "lookup_index"  [attr-defined]
src/inmanta/ast/statements/assign.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/ast/statements/assign.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/ast/statements/assign.py:0: error: Argument 3 to "ExecutionUnit" has incompatible type "ListLiteral"; expected "ResultVariable[object]"  [arg-type]
src/inmanta/ast/statements/assign.py:0: error: Argument 3 to "requires_emit_gradual" of "ExpressionStatement" has incompatible type "ListLiteral"; expected "ResultCollector[object]"  [arg-type]
src/inmanta/ast/statements/assign.py:0: error: Argument 4 to "SetAttributeHelper" has incompatible type "dict[object, VariableABC[Any]]"; expected "dict[object, ResultVariable[Any]]"  [arg-type]
src/inmanta/ast/statements/assign.py:0: error: Argument 5 to "__init__" of "ExecutionUnit" has incompatible type "dict[object, ResultVariable[Any]]"; expected "dict[object, VariableABC[Any]]"  [arg-type]
src/inmanta/ast/statements/assign.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "Type")  [assignment]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "ResultCollector"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/assign.py:0: error: Need type annotation for "temp"  [var-annotated]
src/inmanta/ast/statements/assign.py:0: note: "dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
src/inmanta/ast/statements/assign.py:0: note: Consider using "Mapping" instead, which is covariant in the value type
src/inmanta/ast/statements/call.py:0: error: "Plugin" has no attribute "opts"  [attr-defined]
src/inmanta/ast/statements/call.py:0: error: "Plugin" has no attribute "opts"  [attr-defined]
src/inmanta/ast/statements/call.py:0: error: "Plugin" has no attribute "opts"  [attr-defined]
src/inmanta/ast/statements/call.py:0: error: "Plugin" has no attribute "opts"  [attr-defined]
src/inmanta/ast/statements/call.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/ast/statements/call.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/ast/statements/call.py:0: error: Item "None" of "Function | None" has no attribute "call_direct"  [union-attr]
src/inmanta/ast/statements/call.py:0: error: Item "None" of "Function | None" has no attribute "call_in_context"  [union-attr]
src/inmanta/ast/statements/call.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/call.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/call.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/call.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/call.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/statements/call.py:0: error: Need type annotation for "temp"  [var-annotated]
src/inmanta/ast/statements/define.py:0: error: Argument 1 to "HyphenException" has incompatible type "LocatableString | None"; expected "LocatableString"  [arg-type]
src/inmanta/ast/statements/define.py:0: error: Argument 1 to "HyphenException" has incompatible type "LocatableString | None"; expected "LocatableString"  [arg-type]
src/inmanta/ast/statements/define.py:0: error: Argument 1 to "set_multiplicity" of "RelationAttribute" has incompatible type "tuple[int, int | None] | None"; expected "tuple[int, int | None]"  [arg-type]
src/inmanta/ast/statements/define.py:0: error: Argument 1 to "set_multiplicity" of "RelationAttribute" has incompatible type "tuple[int, int | None] | None"; expected "tuple[int, int | None]"  [arg-type]
src/inmanta/ast/statements/define.py:0: error: Argument 2 to "AttributeAnchor" has incompatible type "Attribute | None"; expected "Attribute"  [arg-type]
src/inmanta/ast/statements/define.py:0: error: Incompatible types in assignment (expression has type "Namespace | None", variable has type "Namespace")  [assignment]
src/inmanta/ast/statements/define.py:0: error: Incompatible types in assignment (expression has type "property", variable has type "ExpressionStatement")  [assignment]
src/inmanta/ast/statements/define.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/define.py:0: error: Return type "None" of "register_types" incompatible with return type "tuple[str, NamedType]" in supertype "inmanta.ast.statements.TypeDefinitionStatement"  [override]
src/inmanta/ast/statements/generator.py:0: error: "Constructor" has no attribute "get_default_values"  [attr-defined]
src/inmanta/ast/statements/generator.py:0: error: Argument 1 to "TypingException" has incompatible type "LocatableString"; expected "Locatable | None"  [arg-type]
src/inmanta/ast/statements/generator.py:0: error: Argument 1 to "TypingException" has incompatible type "LocatableString"; expected "Locatable | None"  [arg-type]
src/inmanta/ast/statements/generator.py:0: error: Argument 1 to "copy_location" of "Locatable" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/ast/statements/generator.py:0: error: Argument 2 of "execute" is incompatible with supertype "inmanta.ast.statements.RequiresEmitStatement"; supertype defines the argument type as "Resolver"  [override]
src/inmanta/ast/statements/generator.py:0: error: Argument 2 to "DuplicateException" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/ast/statements/generator.py:0: error: Argument 4 to "SetAttributeHelper" has incompatible type "dict[object, VariableABC[Any]]"; expected "dict[object, ResultVariable[Any]]"  [arg-type]
src/inmanta/ast/statements/generator.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/generator.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/ast/statements/generator.py:0: error: Name "Dict" is not defined  [name-defined]
src/inmanta/ast/statements/generator.py:0: error: Name "Dict" is not defined  [name-defined]
src/inmanta/ast/statements/generator.py:0: error: Name "Dict" is not defined  [name-defined]
src/inmanta/ast/statements/generator.py:0: error: No overload variant of "__ror__" of "dict" matches argument type "Mapping[str, object]"  [operator]
src/inmanta/ast/statements/generator.py:0: error: Returning Any from function declared to return "dict[str, ExpressionStatement]"  [no-any-return]
src/inmanta/ast/statements/generator.py:0: note: Did you forget to import it from "typing"? (Suggestion: "from typing import Dict")
src/inmanta/ast/statements/generator.py:0: note: Did you forget to import it from "typing"? (Suggestion: "from typing import Dict")
src/inmanta/ast/statements/generator.py:0: note: Did you forget to import it from "typing"? (Suggestion: "from typing import Dict")
src/inmanta/ast/statements/generator.py:0: note: Possible overload variants:
src/inmanta/ast/statements/generator.py:0: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
src/inmanta/ast/statements/generator.py:0: note: This violates the Liskov substitution principle
src/inmanta/ast/statements/generator.py:0: note: def [_T1, _T2] __ror__(self, dict[_T1, _T2], /) -> dict[str | _T1, object]
src/inmanta/ast/statements/generator.py:0: note: def __ror__(self, dict[str, object], /) -> dict[str, object]
src/inmanta/ast/type.py:0: error: "List" expects no type arguments, but 1 given  [type-arg]
src/inmanta/ast/type.py:0: error: "Type" has no attribute "element_type"  [attr-defined]
src/inmanta/ast/type.py:0: error: "Type" has no attribute "get_paired_dataclass"  [attr-defined]
src/inmanta/ast/type.py:0: error: "inmanta.ast.type.List" has no attribute "append"  [attr-defined]
src/inmanta/ast/type.py:0: error: "inmanta.ast.type.List" has no attribute "append"  [attr-defined]
src/inmanta/ast/type.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/ast/type.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/ast/type.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/ast/type.py:0: error: Incompatible types in assignment (expression has type "list[Never]", variable has type "inmanta.ast.type.List")  [assignment]
src/inmanta/ast/type.py:0: error: Incompatible types in assignment (expression has type "type[float]", variable has type "Callable[[object], object]")  [assignment]
src/inmanta/ast/type.py:0: error: Incompatible types in assignment (expression has type "type[float]", variable has type "Callable[[object], object]")  [assignment]
src/inmanta/ast/type.py:0: error: Incompatible types in assignment (expression has type "type[int]", variable has type "Callable[[object], object]")  [assignment]
src/inmanta/ast/type.py:0: error: Item "Locatable" of "Locatable | None" has no attribute "name"  [union-attr]
src/inmanta/ast/type.py:0: error: Item "None" of "Locatable | None" has no attribute "name"  [union-attr]
src/inmanta/ast/type.py:0: error: No overload variant of "reversed" matches argument type "inmanta.ast.type.List"  [call-overload]
src/inmanta/ast/type.py:0: error: Return type "str | None" of "type_string" incompatible with return type "str" in supertype "List"  [override]
src/inmanta/ast/type.py:0: note: Possible overload variants:
src/inmanta/ast/type.py:0: note: def [_T] __new__(cls, Reversible[_T], /) -> reversed[_T]
src/inmanta/ast/type.py:0: note: def [_T] __new__(cls, SupportsLenAndGetItem[_T], /) -> reversed[_T]
src/inmanta/ast/variables.py:0: error: Argument 1 to "Reference" has incompatible type "str"; expected "LocatableString"  [arg-type]
src/inmanta/ast/variables.py:0: error: Argument 1 to "set_value" of "ResultVariable" has incompatible type "bool | Unknown"; expected "bool"  [arg-type]
src/inmanta/ast/variables.py:0: error: Argument 1 to "set_value" of "ResultVariable" has incompatible type "bool | Unknown"; expected "bool"  [arg-type]
src/inmanta/ast/variables.py:0: error: Argument 2 to "AttributeReference" has incompatible type "str"; expected "LocatableString"  [arg-type]
src/inmanta/ast/variables.py:0: error: Incompatible types in assignment (expression has type "Type | ResultVariable[Any]", target has type "VariableABC[Any]")  [assignment]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "ResultCollector"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "ResultCollector"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/ast/variables.py:0: error: Missing type parameters for generic type "VariableReader"  [type-arg]
src/inmanta/compiler/__init__.py:0: error: Incompatible types in assignment (expression has type "Namespace | None", variable has type "Namespace")  [assignment]
src/inmanta/compiler/__init__.py:0: error: Incompatible types in assignment (expression has type "Namespace | None", variable has type "Namespace")  [assignment]
src/inmanta/compiler/__init__.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/compiler/__init__.py:0: error: Module "inmanta.ast" does not explicitly export attribute "Statement"  [attr-defined]
src/inmanta/compiler/__init__.py:0: error: Module "inmanta.ast" has no attribute "BasicBlock"  [attr-defined]
src/inmanta/compiler/help/explainer.py:0: error: Item "None" of "Reference | None" has no attribute "pretty_print"  [union-attr]
src/inmanta/compiler/help/explainer.py:0: error: Item "None" of "RelationAttribute | None" has no attribute "get_name"  [union-attr]
src/inmanta/compiler/help/explainer.py:0: error: Missing return statement  [return]
src/inmanta/config.py:0: error: "_validate_value_types" undefined in superclass  [misc]
src/inmanta/config.py:0: error: Argument 1 to "CronTab" has incompatible type "str | int"; expected "str"  [arg-type]
src/inmanta/config.py:0: error: Cannot infer type argument 1 of "Option"  [misc]
src/inmanta/config.py:0: error: Incompatible default for argument "validator" (default has type "Callable[[str], str]", argument has type "Callable[[str | T], T]")  [assignment]
src/inmanta/config.py:0: error: Missing type parameters for generic type "Option"  [type-arg]
src/inmanta/config.py:0: error: Missing type parameters for generic type "Option"  [type-arg]
src/inmanta/config.py:0: error: Missing type parameters for generic type "Option"  [type-arg]
src/inmanta/config.py:0: error: Missing type parameters for generic type "Option"  [type-arg]
src/inmanta/config.py:0: error: Returning Any from function declared to return "T"  [no-any-return]
src/inmanta/data/__init__.py:0: error: "BaseDocument" has no attribute "__name__"  [attr-defined]
src/inmanta/data/__init__.py:0: error: "BaseDocument" has no attribute "__name__"  [attr-defined]
src/inmanta/data/__init__.py:0: error: "BaseDocument" has no attribute "_fields"; maybe "load_fields"?  [attr-defined]
src/inmanta/data/__init__.py:0: error: "None" not callable  [misc]
src/inmanta/data/__init__.py:0: error: "_validate_single" of "Field" does not return a value (it only ever returns None)  [func-returns-value]
src/inmanta/data/__init__.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/data/__init__.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/data/__init__.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/data/__init__.py:0: error: "type[TBaseDocument]" has no attribute "get_list_without_details"  [attr-defined]
src/inmanta/data/__init__.py:0: error: "type[object]" has no attribute "__supertype__"  [attr-defined]
src/inmanta/data/__init__.py:0: error: Argument "agent" to "ReleasedResourceDetails" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "attributes" to "ReleasedResourceDetails" has incompatible type "object"; expected "dict[str, Any]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "auth_method" to "UserWithRoles" has incompatible type "object"; expected "AuthMethod"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "changes" to "ResourceAction" has incompatible type "dict[ResourceVersionIdStr, dict[str, object]] | None"; expected "dict[str, Any] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "columns" to "copy_records_to_table" of "Connection" has incompatible type "KeysView[str]"; expected "list[str] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "command" to "CompileRunReport" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "compile_data" to "CompileDetails" has incompatible type "object"; expected "CompileData | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "completed" to "CompileDetails" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "completed" to "CompileRunReport" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "default" to "EnvironmentSetting" has incompatible type "int | float | str | dict[str, str | int | bool] | None"; expected "bool | int | float | str | dict[str, str | int | bool]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "do_export" to "CompileDetails" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "doc" to "EnvironmentSetting" has incompatible type "str | None"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "environment" to "CompileDetails" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "environment_variables" to "CompileDetails" has incompatible type "object"; expected "dict[str, str]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "environment_variables" to "CompileRun" has incompatible type "dict[str, str] | None"; expected "dict[str, str]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "errstream" to "CompileRunReport" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "exporter_plugin" to "CompileDetails" has incompatible type "object"; expected "str | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "failed_compile_message" to "CompileDetails" has incompatible type "object"; expected "str | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "first_generated_time" to "ReleasedResourceDetails" has incompatible type "object"; expected "datetime"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "first_generated_version" to "ReleasedResourceDetails" has incompatible type "object"; expected "int"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "force_update" to "CompileDetails" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "id" to "CompileDetails" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "id" to "CompileRunReport" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "id_attribute_value" to "ReleasedResourceDetails" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "is_admin" to "UserWithRoles" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "last_deploy" to "ReleasedResourceDetails" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "mergeable_environment_variables" to "CompileDetails" has incompatible type "object"; expected "dict[str, str]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "metadata" to "CompileDetails" has incompatible type "object"; expected "dict[str, Any]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "name" to "CompileRunReport" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "notify_failed_compile" to "CompileDetails" has incompatible type "object"; expected "bool | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "outstream" to "CompileRunReport" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "partial" to "CompileDetails" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "remote_id" to "CompileDetails" has incompatible type "object"; expected "UUID | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "removed_resource_sets" to "CompileDetails" has incompatible type "object"; expected "list[str]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "requested" to "CompileDetails" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "requested_environment_variables" to "CompileDetails" has incompatible type "object"; expected "dict[str, str]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "resource_id" to "ReleasedResourceDetails" has incompatible type "object"; expected "ResourceIdStr"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "resource_type" to "ReleasedResourceDetails" has incompatible type "object"; expected "ResourceType"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "returncode" to "CompileRunReport" has incompatible type "object"; expected "int | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "started" to "CompileDetails" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "started" to "CompileRunReport" has incompatible type "object"; expected "datetime"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "status" to "ReleasedResourceDetails" has incompatible type "object"; expected "ReleasedResourceState"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "success" to "CompileDetails" has incompatible type "object"; expected "bool | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "username" to "UserWithRoles" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[bool | str], bool]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[bool | str], bool]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[bool | str], bool]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[bool | str], bool]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[float | int | str], float]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[float | int | str], int | float]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[float | int | str], int | float]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[float | int | str], int | float]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[float | int | str], int | float]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[int | str], str]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[int | str], str]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "validator" to "Setting" has incompatible type "Callable[[str, bool], str]"; expected "Callable[[bool | int | float | str | dict[str, str | int | bool]], bool | int | float | str | dict[str, str | int | bool]] | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument "version" to "CompileDetails" has incompatible type "object"; expected "int | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 of "insert_many" is incompatible with supertype "BaseDocument"; supertype defines the argument type as "Sequence[BaseDocument]"  [override]
src/inmanta/data/__init__.py:0: error: Argument 1 to "__init__" of "BaseDocument" has incompatible type "**dict[str, object]"; expected "bool"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "__mangle_dict" of "Resource" has incompatible type "Record"; expected "dict[Any, Any]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "_execute_query" of "BaseDocument" has incompatible type "dict[str, object]"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "_get_composed_filter" of "BaseDocument" has incompatible type "**dict[str, object]"; expected "int"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "_get_composed_filter" of "BaseDocument" has incompatible type "**dict[str, object]"; expected "int"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "_get_composed_filter" of "BaseDocument" has incompatible type "**dict[str, object]"; expected "int"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "_get_composed_filter" of "BaseDocument" has incompatible type "**dict[str, object]"; expected "str | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "_get_composed_filter" of "BaseDocument" has incompatible type "**dict[str, object]"; expected "str | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "_get_composed_filter" of "BaseDocument" has incompatible type "**dict[str, object]"; expected "str | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "add_logs" of "ResourceAction" has incompatible type "list[dict[str, object]]"; expected "str | None"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "append" of "list" has incompatible type "ConfigurationModel"; expected "dict[str, object]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "append" of "list" has incompatible type "Record"; expected "Resource"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "append" of "list" has incompatible type "Record"; expected "dict[str, object]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "append" of "list" has incompatible type "Resource"; expected "dict[str, object]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "append" of "list" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "append" of "list" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "create_from_db_result" of "ResourceDeploySummary" has incompatible type "dict[object, object]"; expected "dict[str, int]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 1 to "parse_id" of "Id" has incompatible type "object"; expected "ResourceVersionIdStr | ResourceIdStr"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 2 to "get_range_filter" of "BaseDocument" has incompatible type "object"; expected "Sequence[tuple[RangeOperator, datetime]] | Sequence[tuple[RangeOperator, int]]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 2 to "is_sub_dict" has incompatible type "dict[str, object]"; expected "dict[UUID | bool | int | float | datetime | str | None, UUID | bool | int | float | datetime | str | None]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Argument 2 to "set" of "Environment" has incompatible type "int | float | str | dict[str, str | int | bool] | None"; expected "bool | int | float | str | dict[str, str | int | bool]"  [arg-type]
src/inmanta/data/__init__.py:0: error: Enum index should be a string (actual index type "object")  [misc]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "Agent | None", expected "Agent")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "ResourceAction | None", expected "ResourceAction")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "ResourceAction | None", expected "ResourceAction")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "Sequence[Parameter]", expected "list[Parameter]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "Sequence[Parameter]", expected "list[Parameter]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "Sequence[Record]", expected "list[TBaseDocument]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "Sequence[Record]", expected "list[TBaseDocument]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "Sequence[Resource]", expected "list[Resource]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "list[Resource] | list[dict[str, object]]", expected "list[Resource]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "list[dict[str, object]]", expected "list[ConfigurationModel]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "list[object]", expected "list[str]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "object", expected "T")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "object", expected "list[Any]")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "object", expected "str | int | bool | datetime | UUID | None")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible return value type (got "object", expected "str")  [return-value]
src/inmanta/data/__init__.py:0: error: Incompatible types in "await" (actual type "AsyncContextManager[Connection]", expected type "Awaitable[Any]")  [misc]
src/inmanta/data/__init__.py:0: error: Incompatible types in "await" (actual type "Pool", expected type "Awaitable[Any]")  [misc]
src/inmanta/data/__init__.py:0: error: Incompatible types in assignment (expression has type "DocumentMeta", variable has type "type[BaseDocument]")  [assignment]
src/inmanta/data/__init__.py:0: error: Incompatible types in assignment (expression has type "dict[str, object]", variable has type "Record")  [assignment]
src/inmanta/data/__init__.py:0: error: Incompatible types in assignment (expression has type "object", variable has type "int")  [assignment]
src/inmanta/data/__init__.py:0: error: Incompatible types in assignment (expression has type "object", variable has type "str")  [assignment]
src/inmanta/data/__init__.py:0: error: Incompatible types in assignment (expression has type "str", variable has type "ResourceIdStr")  [assignment]
src/inmanta/data/__init__.py:0: error: Incompatible types in assignment (expression has type "str", variable has type "ResourceIdStr")  [assignment]
src/inmanta/data/__init__.py:0: error: Incompatible types in assignment (expression has type "str", variable has type "dict[str, object]")  [assignment]
src/inmanta/data/__init__.py:0: error: Invalid index type "object" for "dict[UUID, list[str]]"; expected type "UUID"  [index]
src/inmanta/data/__init__.py:0: error: Invalid index type "object" for "dict[UUID, list[str]]"; expected type "UUID"  [index]
src/inmanta/data/__init__.py:0: error: Invalid index type "object" for "dict[UUID, list[str]]"; expected type "UUID"  [index]
src/inmanta/data/__init__.py:0: error: Invalid index type "object" for "dict[UUID, list[str]]"; expected type "UUID"  [index]
src/inmanta/data/__init__.py:0: error: Item "None" of "RowLockMode | None" has no attribute "value"  [union-attr]
src/inmanta/data/__init__.py:0: error: Key expression in dictionary comprehension has incompatible type "object"; expected type "ResourceIdStr"  [misc]
src/inmanta/data/__init__.py:0: error: Key expression in dictionary comprehension has incompatible type "object"; expected type "ResourceIdStr"  [misc]
src/inmanta/data/__init__.py:0: error: Key expression in dictionary comprehension has incompatible type "object"; expected type "ResourceIdStr"  [misc]
src/inmanta/data/__init__.py:0: error: Key expression in dictionary comprehension has incompatible type "str"; expected type "ResourceIdStr"  [misc]
src/inmanta/data/__init__.py:0: error: List comprehension has incompatible type List[object]; expected List[str]  [misc]
src/inmanta/data/__init__.py:0: error: List comprehension has incompatible type List[object]; expected List[str]  [misc]
src/inmanta/data/__init__.py:0: error: List item 0 has incompatible type "object"; expected "str"  [list-item]
src/inmanta/data/__init__.py:0: error: List item 0 has incompatible type "object"; expected "str"  [list-item]
src/inmanta/data/__init__.py:0: error: Missing type parameters for generic type "Field"  [type-arg]
src/inmanta/data/__init__.py:0: error: Missing type parameters for generic type "Field"  [type-arg]
src/inmanta/data/__init__.py:0: error: Missing type parameters for generic type "Field"  [type-arg]
src/inmanta/data/__init__.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/data/__init__.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/data/__init__.py:0: error: Missing type parameters for generic type "list"  [type-arg]
src/inmanta/data/__init__.py:0: error: Need type annotation for "_updates" (hint: "_updates: dict[<type>, <type>] = ...")  [var-annotated]
src/inmanta/data/__init__.py:0: error: No overload variant of "__add__" of "list" matches argument type "Sequence[tuple[str, ColumnType, PagingOrder]]"  [operator]
src/inmanta/data/__init__.py:0: error: No overload variant of "int" matches argument type "object"  [call-overload]
src/inmanta/data/__init__.py:0: error: No overload variant of "int" matches argument type "object"  [call-overload]
src/inmanta/data/__init__.py:0: error: No overload variant of "list" matches argument type "object"  [call-overload]
src/inmanta/data/__init__.py:0: error: Overloaded function implementation does not accept all possible arguments of signature 1  [misc]
src/inmanta/data/__init__.py:0: error: Returning Any from function declared to return "Connection"  [no-any-return]
src/inmanta/data/__init__.py:0: error: Returning Any from function declared to return "Pool"  [no-any-return]
src/inmanta/data/__init__.py:0: error: Returning Any from function declared to return "int | None"  [no-any-return]
src/inmanta/data/__init__.py:0: error: Returning Any from function declared to return "list[TBaseDocument]"  [no-any-return]
src/inmanta/data/__init__.py:0: error: Returning Any from function declared to return "list[tuple[ColumnNameStr, ColumnType, PagingOrder]]"  [no-any-return]
src/inmanta/data/__init__.py:0: error: Signature of "_generate_next_value_" incompatible with supertype "enum.Enum"  [override]
src/inmanta/data/__init__.py:0: error: Signature of "get_list" incompatible with supertype "BaseDocument"  [override]
src/inmanta/data/__init__.py:0: error: Unsupported left operand type for + ("dict[str, object]")  [operator]
src/inmanta/data/__init__.py:0: error: Unsupported left operand type for + ("object")  [operator]
src/inmanta/data/__init__.py:0: error: Unsupported right operand type for in ("object")  [operator]
src/inmanta/data/__init__.py:0: error: Unsupported target for indexed assignment ("object")  [index]
src/inmanta/data/__init__.py:0: error: Unused "type: ignore" comment  [unused-ignore]
src/inmanta/data/__init__.py:0: error: Value expression in dictionary comprehension has incompatible type "object"; expected type "ReleasedResourceState"  [misc]
src/inmanta/data/__init__.py:0: error: Value of type "Record | None" is not indexable  [index]
src/inmanta/data/__init__.py:0: error: Value of type "object" is not indexable  [index]
src/inmanta/data/__init__.py:0: note: @classmethod
src/inmanta/data/__init__.py:0: note: @classmethod
src/inmanta/data/__init__.py:0: note: @staticmethod
src/inmanta/data/__init__.py:0: note: Possible overload variants:
src/inmanta/data/__init__.py:0: note: Possible overload variants:
src/inmanta/data/__init__.py:0: note: Possible overload variants:
src/inmanta/data/__init__.py:0: note: Possible overload variants:
src/inmanta/data/__init__.py:0: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
src/inmanta/data/__init__.py:0: note: Subclass:
src/inmanta/data/__init__.py:0: note: Subclass:
src/inmanta/data/__init__.py:0: note: Superclass:
src/inmanta/data/__init__.py:0: note: Superclass:
src/inmanta/data/__init__.py:0: note: This violates the Liskov substitution principle
src/inmanta/data/__init__.py:0: note: def [_S] __add__(self, list[_S], /) -> list[_S | tuple[str, ColumnType, PagingOrder]]
src/inmanta/data/__init__.py:0: note: def [_T] __init__(self) -> list[_T]
src/inmanta/data/__init__.py:0: note: def [_T] __init__(self, Iterable[_T], /) -> list[_T]
src/inmanta/data/__init__.py:0: note: def __add__(self, list[tuple[str, ColumnType, PagingOrder]], /) -> list[tuple[str, ColumnType, PagingOrder]]
src/inmanta/data/__init__.py:0: note: def __new__(cls, str | Buffer | SupportsInt | SupportsIndex | SupportsTrunc = ..., /) -> int
src/inmanta/data/__init__.py:0: note: def __new__(cls, str | Buffer | SupportsInt | SupportsIndex | SupportsTrunc = ..., /) -> int
src/inmanta/data/__init__.py:0: note: def __new__(cls, str | bytes | bytearray, /, base: SupportsIndex) -> int
src/inmanta/data/__init__.py:0: note: def __new__(cls, str | bytes | bytearray, /, base: SupportsIndex) -> int
src/inmanta/data/__init__.py:0: note: def _generate_next_value_(name, start: int, count: int, last_values: Sequence[object]) -> str
src/inmanta/data/__init__.py:0: note: def _generate_next_value_(name: str, start: int, count: int, last_values: list[Any]) -> Any
src/inmanta/data/__init__.py:0: note: def get_list(cls, *, order_by_column: str | None = ..., order: str | None = ..., limit: int | None = ..., offset: int | None = ..., no_obj: bool | None = ..., lock: RowLockMode | None = ..., connection: Connection | None = ..., **query: object) -> Coroutine[Any, Any, list[Environment]]
src/inmanta/data/__init__.py:0: note: def get_list(cls, *, order_by_column: str | None = ..., order: str | None = ..., limit: int | None = ..., offset: int | None = ..., no_obj: bool | None = ..., lock: RowLockMode | None = ..., connection: Connection | None = ..., details: bool = ..., **query: object) -> Coroutine[Any, Any, list[Environment]]
src/inmanta/data/dataview.py:0: error: "object" has no attribute "get"  [attr-defined]
src/inmanta/data/dataview.py:0: error: "object" has no attribute "get"  [attr-defined]
src/inmanta/data/dataview.py:0: error: "object" has no attribute "get"  [attr-defined]
src/inmanta/data/dataview.py:0: error: "object" has no attribute "get"  [attr-defined]
src/inmanta/data/dataview.py:0: error: "object" has no attribute "get"  [attr-defined]
src/inmanta/data/dataview.py:0: error: "object" has no attribute "get"  [attr-defined]
src/inmanta/data/dataview.py:0: error: "object" has no attribute "get"  [attr-defined]
src/inmanta/data/dataview.py:0: error: Argument "action" to "ResourceLog" has incompatible type "object"; expected "ResourceAction"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "action_id" to "ResourceLog" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "attribute_hash" to "ResourceHistory" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "cleared" to "Notification" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "compile_id" to "Notification" has incompatible type "object"; expected "UUID | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "completed" to "CompileReport" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "created" to "Notification" has incompatible type "object"; expected "datetime"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "date" to "DesiredStateVersion" has incompatible type "object"; expected "datetime"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "date" to "ResourceHistory" has incompatible type "object"; expected "datetime"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "discovered_resource_id" to "DiscoveredResource" has incompatible type "object"; expected "ResourceIdStr"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "discovery_resource_id" to "DiscoveredResource" has incompatible type "object"; expected "ResourceIdStr | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "do_export" to "CompileReport" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "environment" to "Agent" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "environment" to "CompileReport" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "environment" to "Fact" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "environment" to "Notification" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "environment" to "Parameter" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "environment_variables" to "CompileReport" has incompatible type "object"; expected "dict[str, str]"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "exporter_plugin" to "CompileReport" has incompatible type "object"; expected "str | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "failed_compile_message" to "CompileReport" has incompatible type "object"; expected "str | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "force_update" to "CompileReport" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "id" to "CompileReport" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "id" to "Fact" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "id" to "Notification" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "id" to "Parameter" has incompatible type "object"; expected "UUID"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "last_failover" to "Agent" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "mergeable_environment_variables" to "CompileReport" has incompatible type "object"; expected "dict[str, str]"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "message" to "DesiredStateLabel" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "message" to "Notification" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "metadata" to "CompileReport" has incompatible type "object"; expected "dict[str, Any]"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "metadata" to "Fact" has incompatible type "object"; expected "dict[str, Any] | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "metadata" to "Parameter" has incompatible type "object"; expected "dict[str, Any] | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "name" to "Agent" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "name" to "DesiredStateLabel" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "name" to "Fact" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "name" to "Parameter" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "notify_failed_compile" to "CompileReport" has incompatible type "object"; expected "bool | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "partial" to "CompileReport" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "paused" to "Agent" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "process_id" to "Agent" has incompatible type "object"; expected "UUID | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "process_name" to "Agent" has incompatible type "object"; expected "str | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "read" to "Notification" has incompatible type "object"; expected "bool"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "remote_id" to "CompileReport" has incompatible type "object"; expected "UUID | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "removed_resource_sets" to "CompileReport" has incompatible type "object"; expected "list[str]"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "requested" to "CompileReport" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "requested_environment_variables" to "CompileReport" has incompatible type "object"; expected "dict[str, str]"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "resource_id" to "Fact" has incompatible type "object"; expected "ResourceIdStr"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "resource_id" to "LatestReleasedResource" has incompatible type "object"; expected "ResourceIdStr"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "resource_id" to "VersionedResource" has incompatible type "object"; expected "ResourceIdStr"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "severity" to "Notification" has incompatible type "object"; expected "NotificationSeverity"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "source" to "Fact" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "source" to "Parameter" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "started" to "CompileReport" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "status" to "Agent" has incompatible type "object"; expected "AgentStatus"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "status" to "DesiredStateVersion" has incompatible type "object"; expected "DesiredStateVersionStatus"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "status" to "LatestReleasedResource" has incompatible type "object"; expected "ReleasedResourceState"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "success" to "CompileReport" has incompatible type "object"; expected "bool | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "timestamp" to "ResourceLog" has incompatible type "object"; expected "datetime"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "title" to "Notification" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "total" to "DesiredStateVersion" has incompatible type "object"; expected "int"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "unpause_on_resume" to "Agent" has incompatible type "object"; expected "bool | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "updated" to "Fact" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "updated" to "Parameter" has incompatible type "object"; expected "datetime | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "uri" to "Notification" has incompatible type "object"; expected "str | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "value" to "Fact" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "value" to "Parameter" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "values" to "DiscoveredResource" has incompatible type "object"; expected "dict[str, object]"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "version" to "CompileReport" has incompatible type "object"; expected "int | None"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument "version" to "DesiredStateVersion" has incompatible type "object"; expected "int"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument 1 to "get_details_from_resource_id" of "Resource" has incompatible type "object"; expected "ResourceIdStr"  [arg-type]
src/inmanta/data/dataview.py:0: error: Argument 1 to "get_details_from_resource_id" of "Resource" has incompatible type "object"; expected "ResourceIdStr"  [arg-type]
src/inmanta/data/dataview.py:0: error: Incompatible return value type (got "object", expected "dict[str, Any]")  [return-value]
src/inmanta/data/dataview.py:0: error: Return type "Sequence[dict[str, str]]" of "construct_dtos" incompatible with return type "Sequence[DiscoveredResource]" in supertype "DataView"  [override]
src/inmanta/data/dataview.py:0: error: Unsupported left operand type for + ("object")  [operator]
src/inmanta/data/dataview.py:0: error: Unsupported left operand type for + ("object")  [operator]
src/inmanta/data/dataview.py:0: error: Unsupported right operand type for in ("object")  [operator]
src/inmanta/data/dataview.py:0: error: Unsupported target for indexed assignment ("object")  [index]
src/inmanta/data/model.py:0: error: Argument 1 of "__setstate__" is incompatible with supertype "pydantic.main.BaseModel"; supertype defines the argument type as "dict[Any, Any]"  [override]
src/inmanta/data/model.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/data/model.py:0: error: Return type "str" of "__getstate__" incompatible with return type "dict[Any, Any]" in supertype "pydantic.main.BaseModel"  [override]
src/inmanta/data/model.py:0: error: Unused "type: ignore" comment, use narrower [prop-decorator] instead of [misc] code  [unused-ignore]
src/inmanta/data/model.py:0: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
src/inmanta/data/model.py:0: note: This violates the Liskov substitution principle
src/inmanta/data/schema.py:0: error: No overload variant of "set" matches argument type "object"  [call-overload]
src/inmanta/data/schema.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/data/schema.py:0: error: Returning Any from function declared to return "set[int]"  [no-any-return]
src/inmanta/data/schema.py:0: note: Possible overload variants:
src/inmanta/data/schema.py:0: note: def [_T] __init__(self) -> set[_T]
src/inmanta/data/schema.py:0: note: def [_T] __init__(self, Iterable[_T], /) -> set[_T]
src/inmanta/deploy/scheduler.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/deploy/state.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/deploy/state.py:0: error: Argument "attribute_hash" to "ResourceIntent" has incompatible type "object"; expected "str"  [arg-type]
src/inmanta/deploy/state.py:0: error: Argument "attributes" to "ResourceIntent" has incompatible type "object"; expected "Mapping[str, object]"  [arg-type]
src/inmanta/deploy/state.py:0: error: Enum index should be a string (actual index type "object")  [misc]
src/inmanta/deploy/state.py:0: error: Enum index should be a string (actual index type "object")  [misc]
src/inmanta/deploy/state.py:0: error: Enum index should be a string (actual index type "object")  [misc]
src/inmanta/deploy/state.py:0: error: Enum index should be a string (actual index type "object")  [misc]
src/inmanta/deploy/state.py:0: error: Unsupported left operand type for > ("object")  [operator]
src/inmanta/docstring_parser.py:0: error: Argument 1 to "DocString" has incompatible type "str"; expected "Docstring"  [arg-type]
src/inmanta/docstring_parser.py:0: error: Incompatible types in assignment (expression has type "Docstring", variable has type "str")  [assignment]
src/inmanta/docstring_parser.py:0: error: Value expression in dictionary comprehension has incompatible type "str | None"; expected type "str"  [misc]
src/inmanta/env.py:0: error: Argument 1 to "parse_requirements" has incompatible type "Sequence[CanonicalRequirement]"; expected "Sequence[str]"  [arg-type]
src/inmanta/env.py:0: error: Incompatible return value type (got "tuple[str | None, Loader | None] | None", expected "tuple[str | None, Loader] | None")  [return-value]
src/inmanta/env.py:0: error: Item "None" of "ModuleSpec | None" has no attribute "submodule_search_locations"  [union-attr]
src/inmanta/env.py:0: error: Item "None" of "ModuleSpec | None" has no attribute "submodule_search_locations"  [union-attr]
src/inmanta/env.py:0: error: Module has no attribute "real_prefix"  [attr-defined]
src/inmanta/execute/dataflow/__init__.py:0: error: Missing type parameters for generic type "Assignment"  [type-arg]
src/inmanta/execute/dataflow/__init__.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/dataflow/__init__.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/dataflow/__init__.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/dataflow/__init__.py:0: error: No return value expected  [return-value]
src/inmanta/execute/dataflow/datatrace.py:0: error: Missing type parameters for generic type "Assignment"  [type-arg]
src/inmanta/execute/dataflow/datatrace.py:0: error: Missing type parameters for generic type "Assignment"  [type-arg]
src/inmanta/execute/dataflow/graphic.py:0: error: Missing type parameters for generic type "Assignment"  [type-arg]
src/inmanta/execute/dataflow/graphic.py:0: error: Missing type parameters for generic type "Assignment"  [type-arg]
src/inmanta/execute/dataflow/graphic.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/dataflow/graphic.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/proxy.py:0: error: "Instance" not callable  [operator]
src/inmanta/execute/proxy.py:0: error: "Type" has no attribute "from_python"; maybe "to_python"?  [attr-defined]
src/inmanta/execute/proxy.py:0: error: Argument 1 to "AttributeNotFound" has incompatible type "Locatable | None"; expected "Statement | None"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 1 to "DynamicProxy" has incompatible type "object"; expected "Instance"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 1 to "RuntimeException" has incompatible type "DictProxy"; expected "Locatable | None"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 1 to "SequenceProxy" has incompatible type "object"; expected "Sequence[object]"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 1 to "len" has incompatible type "Instance"; expected "Sized"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 1 to "len" has incompatible type "Instance"; expected "Sized"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 2 to "__init__" of "DynamicProxy" has incompatible type "Callable[..., object]"; expected "Instance"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 2 to "__init__" of "DynamicProxy" has incompatible type "Sequence[object]"; expected "Instance"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 2 to "__init__" of "DynamicProxy" has incompatible type "dict[str, object]"; expected "Instance"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Argument 2 to "__init__" of "DynamicProxy" has incompatible type "enumerate[object]"; expected "Instance"  [arg-type]
src/inmanta/execute/proxy.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/execute/proxy.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/execute/proxy.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/execute/proxy.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/execute/proxy.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/execute/proxy.py:0: error: No overload variant of "iter" matches argument type "Instance"  [call-overload]
src/inmanta/execute/proxy.py:0: error: No overload variant of "iter" matches argument type "Instance"  [call-overload]
src/inmanta/execute/proxy.py:0: error: No overload variant of "next" matches argument type "Instance"  [call-overload]
src/inmanta/execute/proxy.py:0: error: Returning Any from function declared to return "Entity | None"  [no-any-return]
src/inmanta/execute/proxy.py:0: error: Returning Any from function declared to return "Instance"  [no-any-return]
src/inmanta/execute/proxy.py:0: error: Returning Any from function declared to return "ProxyContext"  [no-any-return]
src/inmanta/execute/proxy.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/execute/proxy.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/execute/proxy.py:0: error: Unsupported left operand type for < ("Instance")  [operator]
src/inmanta/execute/proxy.py:0: error: Value expression in dictionary comprehension has incompatible type "object"; expected type "UUID | int | float | datetime | str | None"  [misc]
src/inmanta/execute/proxy.py:0: error: Value of type "Instance" is not indexable  [index]
src/inmanta/execute/proxy.py:0: error: Value of type "Instance" is not indexable  [index]
src/inmanta/execute/proxy.py:0: note: Possible overload variants:
src/inmanta/execute/proxy.py:0: note: Possible overload variants:
src/inmanta/execute/proxy.py:0: note: Possible overload variants:
src/inmanta/execute/proxy.py:0: note: def [_SupportsNextT_co: SupportsNext[Any]] iter(SupportsIter[_SupportsNextT_co], /) -> _SupportsNextT_co
src/inmanta/execute/proxy.py:0: note: def [_SupportsNextT_co: SupportsNext[Any]] iter(SupportsIter[_SupportsNextT_co], /) -> _SupportsNextT_co
src/inmanta/execute/proxy.py:0: note: def [_T, _VT] next(SupportsNext[_T], _VT, /) -> _T | _VT
src/inmanta/execute/proxy.py:0: note: def [_T] iter(Callable[[], _T | None], None, /) -> Iterator[_T]
src/inmanta/execute/proxy.py:0: note: def [_T] iter(Callable[[], _T | None], None, /) -> Iterator[_T]
src/inmanta/execute/proxy.py:0: note: def [_T] iter(Callable[[], _T], object, /) -> Iterator[_T]
src/inmanta/execute/proxy.py:0: note: def [_T] iter(Callable[[], _T], object, /) -> Iterator[_T]
src/inmanta/execute/proxy.py:0: note: def [_T] iter(_GetItemIterable[_T], /) -> Iterator[_T]
src/inmanta/execute/proxy.py:0: note: def [_T] iter(_GetItemIterable[_T], /) -> Iterator[_T]
src/inmanta/execute/proxy.py:0: note: def [_T] next(SupportsNext[_T], /) -> _T
src/inmanta/execute/runtime.py:0: error: "BaseListVariable" has no attribute "attribute"  [attr-defined]
src/inmanta/execute/runtime.py:0: error: "BaseListVariable" has no attribute "attribute"  [attr-defined]
src/inmanta/execute/runtime.py:0: error: "BaseListVariable" has no attribute "myself"  [attr-defined]
src/inmanta/execute/runtime.py:0: error: "BaseListVariable" has no attribute "myself"  [attr-defined]
src/inmanta/execute/runtime.py:0: error: Argument 1 to "AttributeException" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 1 to "RuntimeException" has incompatible type "Instance"; expected "Locatable | None"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 1 to "__init__" of "Locatable" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 1 to "get_location" of "Locatable" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 1 to "ready" of "Waiter" has incompatible type "WrappedValueVariable[T]"; expected "ResultVariable[Any] | Waiter"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 1 to "set_location" of "Locatable" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 2 to "UnsetException" has incompatible type "ResultVariable[T]"; expected "Instance | None"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 2 to "assign" of "AssignableNodeReference" has incompatible type "Instance"; expected "Locatable"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 2 to "set_value" of "ResultVariable" has incompatible type "None"; expected "Location"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Argument 4 to "resume" of "Resumer" has incompatible type "ResultVariable[Any] | None"; expected "ResultVariable[Any]"  [arg-type]
src/inmanta/execute/runtime.py:0: error: Incompatible return value type (got "T | None", expected "T")  [return-value]
src/inmanta/execute/runtime.py:0: error: Incompatible types in assignment (expression has type "Attribute", variable has type "RelationAttribute")  [assignment]
src/inmanta/execute/runtime.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "Location")  [assignment]
src/inmanta/execute/runtime.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "QueueScheduler")  [assignment]
src/inmanta/execute/runtime.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "list[Waiter]")  [assignment]
src/inmanta/execute/runtime.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "list[Waiter]")  [assignment]
src/inmanta/execute/runtime.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "list[Waiter]")  [assignment]
src/inmanta/execute/runtime.py:0: error: Incompatible types in assignment (expression has type "object", variable has type "Instance | None")  [assignment]
src/inmanta/execute/runtime.py:0: error: Item "list[Instance]" of "Instance | list[Instance]" has no attribute "set_attribute"  [union-attr]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "DelayedResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/execute/scheduler.py:0: error: "<subclass of "inmanta.execute.runtime.DelayedResultVariable[builtins.object]" and "inmanta.execute.runtime.RelationAttributeVariable">" has no attribute "attribute"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: "<subclass of "inmanta.execute.runtime.DelayedResultVariable[builtins.object]" and "inmanta.execute.runtime.RelationAttributeVariable">" has no attribute "attribute"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: "<subclass of "inmanta.execute.runtime.DelayedResultVariable[builtins.object]" and "inmanta.execute.runtime.RelationAttributeVariable">" has no attribute "attribute"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: "DefinitionStatement" has no attribute "evaluate"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: "Instance" has no attribute "dump"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: "Instance" has no attribute "dump"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: "Type" has no attribute "get_all_instances"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: "Type" has no attribute "get_all_instances"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: "Type" has no attribute "get_all_instances"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: Argument 2 to "requeue_with_additional_requires" of "Waiter" has incompatible type "Instance | None"; expected "ResultVariable[Any]"  [arg-type]
src/inmanta/execute/scheduler.py:0: error: Incompatible return value type (got "list[tuple[Range, AnchorTarget | None]]", expected "Sequence[tuple[Location, AnchorTarget]]")  [return-value]
src/inmanta/execute/scheduler.py:0: error: Missing type parameters for generic type "DelayedResultVariable"  [type-arg]
src/inmanta/execute/scheduler.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/execute/scheduler.py:0: error: Missing type parameters for generic type "VariableABC"  [type-arg]
src/inmanta/execute/scheduler.py:0: error: Module "inmanta.ast" does not explicitly export attribute "NamedType"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: Module "inmanta.ast" does not explicitly export attribute "Statement"  [attr-defined]
src/inmanta/execute/scheduler.py:0: error: Module "inmanta.ast" has no attribute "BasicBlock"  [attr-defined]
src/inmanta/export.py:0: error: Argument 1 to "parse_id" of "Id" has incompatible type "str"; expected "ResourceVersionIdStr | ResourceIdStr"  [arg-type]
src/inmanta/export.py:0: error: Argument 1 to "set_version" of "Resource" has incompatible type "int | None"; expected "int"  [arg-type]
src/inmanta/export.py:0: error: Argument 2 to "add" of "Exporter" has incompatible type "tuple[str, ...]"; expected "list[str]"  [arg-type]
src/inmanta/export.py:0: error: Argument 3 to "create_from_model" of "Resource" has incompatible type "str | tuple[object, ...] | int | float | DynamicProxy | Reference[str | float | int | bool | DataclassInstance | None] | None"; expected "DynamicProxy"  [arg-type]
src/inmanta/export.py:0: error: Cannot infer type argument 1 of "Option"  [misc]
src/inmanta/export.py:0: error: Incompatible return value type (got "int | None", expected "int")  [return-value]
src/inmanta/export.py:0: error: List comprehension has incompatible type List[str | tuple[object, ...] | int | float | DynamicProxy | Reference[str | float | int | bool | DataclassInstance | None] | None]; expected List[dict[str, Sequence[str | tuple[Any, ...] | int | float | bool | DynamicProxy]]]  [misc]
src/inmanta/export.py:0: error: Missing type parameters for generic type "tuple"  [type-arg]
src/inmanta/export.py:0: error: Returning Any from function declared to return "int"  [no-any-return]
src/inmanta/export.py:0: error: Too few arguments  [call-arg]
src/inmanta/export.py:0: error: Unexpected keyword argument "types"  [call-arg]
src/inmanta/export.py:0: error: Value of type "dict[str, Any] | None" is not indexable  [index]
src/inmanta/export.py:0: error: Value of type "dict[str, Any] | None" is not indexable  [index]
src/inmanta/export.py:0: error: Value of type "dict[str, Any] | None" is not indexable  [index]
src/inmanta/export.py:0: error: Value of type "dict[str, Any] | None" is not indexable  [index]
src/inmanta/export.py:0: error: Value of type "dict[str, Any] | None" is not indexable  [index]
src/inmanta/loader.py:0: error: Argument 1 to "getsourcefile" has incompatible type "object"; expected Module | type[Any] | MethodType | FunctionType | TracebackType | FrameType | CodeType | Callable[..., Any]  [arg-type]
src/inmanta/loader.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "PluginModuleFinder")  [assignment]
src/inmanta/loader.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "PluginModuleFinder")  [assignment]
src/inmanta/loader.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/loader.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/loader.py:0: error: No return value expected  [return-value]
src/inmanta/loader.py:0: error: Return type "bytes" of "get_source" incompatible with return type "str | None" in supertype "importlib.abc.InspectLoader"  [override]
src/inmanta/logging.py:0: error: Argument "root_handlers" to "FullLoggingConfig" has incompatible type "object"; expected "list[str] | None"  [arg-type]
src/inmanta/logging.py:0: error: Argument "root_log_level" to "FullLoggingConfig" has incompatible type "object"; expected "int | str | None"  [arg-type]
src/inmanta/logging.py:0: error: Returning Any from function declared to return "dict[str, object]"  [no-any-return]
src/inmanta/logging.py:0: error: Returning Any from function declared to return "dict[str, object]"  [no-any-return]
src/inmanta/main.py:0: error: Argument 1 to "len" has incompatible type "object"; expected "Sized"  [arg-type]
src/inmanta/main.py:0: error: Argument 1 to "len" has incompatible type "object"; expected "Sized"  [arg-type]
src/inmanta/main.py:0: error: Argument 1 to "parse_id" of "Id" has incompatible type "object"; expected "ResourceVersionIdStr | ResourceIdStr"  [arg-type]
src/inmanta/main.py:0: error: Returning Any from function declared to return "dict[str, Any] | None"  [no-any-return]
src/inmanta/main.py:0: error: Value of type "dict[str, Any] | None" is not indexable  [index]
src/inmanta/main.py:0: error: Value of type "dict[str, Any] | None" is not indexable  [index]
src/inmanta/main.py:0: error: Value of type "object" is not indexable  [index]
src/inmanta/model.py:0: error: "Value" has no attribute "to_dict"  [attr-defined]
src/inmanta/model.py:0: error: "Value" has no attribute "to_dict"  [attr-defined]
src/inmanta/model.py:0: error: "list[dict[str, Any]]" has no attribute "items"  [attr-defined]
src/inmanta/model.py:0: error: Call to untyped function "ReferenceValue" in typed context  [no-untyped-call]
src/inmanta/model.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/model.py:0: error: No return value expected  [return-value]
src/inmanta/model.py:0: error: Value expression in dictionary comprehension has incompatible type "None"; expected type "Attribute"  [misc]
src/inmanta/module.py:0: error: Argument 1 to "Metadata" has incompatible type "**Mapping[str, object]"; expected "bool"  [arg-type]
src/inmanta/module.py:0: error: Argument 1 to "Metadata" has incompatible type "**Mapping[str, object]"; expected "str | None"  [arg-type]
src/inmanta/module.py:0: error: Argument 1 to "Metadata" has incompatible type "**Mapping[str, object]"; expected "str"  [arg-type]
src/inmanta/module.py:0: error: Missing type parameters for generic type "ModuleLike"  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type "ModuleLike"  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type "ModuleLike"  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/module.py:0: error: Returning Any from function declared to return "Mapping[str, object]"  [no-any-return]
src/inmanta/moduletool.py:0: error: "Module[Any]" has no attribute "is_editable"  [attr-defined]
src/inmanta/moduletool.py:0: error: Argument 1 to "get_project_for_module" of "ModuleTool" has incompatible type "None"; expected "str"  [arg-type]
src/inmanta/moduletool.py:0: error: Missing type parameters for generic type "ModuleLike"  [type-arg]
src/inmanta/moduletool.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/moduletool.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/moduletool.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/moduletool.py:0: error: Missing type parameters for generic type Module  [type-arg]
src/inmanta/moduletool.py:0: error: Skipping analyzing "cookiecutter.main": module is installed, but missing library stubs or py.typed marker  [import-untyped]
src/inmanta/parser/cache.py:0: error: Argument 1 to "ASTUnpickler" has incompatible type "BufferedReader[_BufferedReaderStream]"; expected "BytesIO"  [arg-type]
src/inmanta/parser/parsetab.py:0: error: Need type annotation for "_lr_action" (hint: "_lr_action: dict[<type>, <type>] = ...")  [var-annotated]
src/inmanta/parser/parsetab.py:0: error: Need type annotation for "_lr_goto" (hint: "_lr_goto: dict[<type>, <type>] = ...")  [var-annotated]
src/inmanta/parser/parsetab.py:0: error: No overload variant of "zip" matches argument types "list[int]", "object"  [call-overload]
src/inmanta/parser/parsetab.py:0: note: Possible overload variants:
src/inmanta/parser/parsetab.py:0: note: def [_T_co, _T1, _T2, _T3, _T4, _T5] __new__(cls, Iterable[_T1], Iterable[_T2], Iterable[_T3], Iterable[_T4], Iterable[_T5], /, *, strict: bool = ...) -> zip[tuple[_T1, _T2, _T3, _T4, _T5]]
src/inmanta/parser/parsetab.py:0: note: def [_T_co, _T1, _T2, _T3, _T4] __new__(cls, Iterable[_T1], Iterable[_T2], Iterable[_T3], Iterable[_T4], /, *, strict: bool = ...) -> zip[tuple[_T1, _T2, _T3, _T4]]
src/inmanta/parser/parsetab.py:0: note: def [_T_co, _T1, _T2, _T3] __new__(cls, Iterable[_T1], Iterable[_T2], Iterable[_T3], /, *, strict: bool = ...) -> zip[tuple[_T1, _T2, _T3]]
src/inmanta/parser/parsetab.py:0: note: def [_T_co, _T1, _T2] __new__(cls, Iterable[_T1], Iterable[_T2], /, *, strict: bool = ...) -> zip[tuple[_T1, _T2]]
src/inmanta/parser/parsetab.py:0: note: def [_T_co, _T1] __new__(cls, Iterable[_T1], /, *, strict: bool = ...) -> zip[tuple[_T1]]
src/inmanta/parser/parsetab.py:0: note: def [_T_co] __new__(cls, *, strict: bool = ...) -> zip[Any]
src/inmanta/parser/parsetab.py:0: note: def [_T_co] __new__(cls, Iterable[Any], Iterable[Any], Iterable[Any], Iterable[Any], Iterable[Any], Iterable[Any], /, *iterables: Iterable[Any], strict: bool = ...) -> zip[tuple[Any, ...]]
src/inmanta/parser/plyInmantaLex.py:0: error: Argument 1 to "LocatableString" has incompatible type "str | LocatableString"; expected "str"  [arg-type]
src/inmanta/parser/plyInmantaLex.py:0: error: Argument 1 to "Reference" has incompatible type "str"; expected "LocatableString"  [arg-type]
src/inmanta/parser/plyInmantaLex.py:0: error: Argument 1 to "float" has incompatible type "str | LocatableString"; expected "str | Buffer | SupportsFloat | SupportsIndex"  [arg-type]
src/inmanta/parser/plyInmantaLex.py:0: error: Argument 1 to "get" of "dict" has incompatible type "str | LocatableString"; expected "str"  [arg-type]
src/inmanta/parser/plyInmantaLex.py:0: error: Argument 1 to "int" has incompatible type "str | LocatableString"; expected "str | Buffer | SupportsInt | SupportsIndex | SupportsTrunc"  [arg-type]
src/inmanta/parser/plyInmantaLex.py:0: error: Argument 1 to "len" has incompatible type "str | LocatableString"; expected "Sized"  [arg-type]
src/inmanta/parser/plyInmantaLex.py:0: error: Incompatible types in assignment (expression has type "Regex", variable has type "str | LocatableString")  [assignment]
src/inmanta/parser/plyInmantaLex.py:0: error: Incompatible types in assignment (expression has type "float", variable has type "str | LocatableString")  [assignment]
src/inmanta/parser/plyInmantaLex.py:0: error: Incompatible types in assignment (expression has type "int", variable has type "str | LocatableString")  [assignment]
src/inmanta/parser/plyInmantaLex.py:0: error: Item "LocatableString" of "str | LocatableString" has no attribute "index"  [union-attr]
src/inmanta/parser/plyInmantaLex.py:0: error: Value of type "str | LocatableString" is not indexable  [index]
src/inmanta/parser/plyInmantaLex.py:0: error: Value of type "str | LocatableString" is not indexable  [index]
src/inmanta/parser/plyInmantaLex.py:0: error: Value of type "str | LocatableString" is not indexable  [index]
src/inmanta/parser/plyInmantaLex.py:0: error: Value of type "str | LocatableString" is not indexable  [index]
src/inmanta/parser/plyInmantaLex.py:0: error: Value of type "str | LocatableString" is not indexable  [index]
src/inmanta/parser/plyInmantaParser.py:0: error: Argument 1 to "DefineImport" has incompatible type "str"; expected "LocatableString"  [arg-type]
src/inmanta/parser/plyInmantaParser.py:0: error: Argument 1 to "DefineImport" has incompatible type "str"; expected "LocatableString"  [arg-type]
src/inmanta/parser/plyInmantaParser.py:0: error: Call to untyped function "Not" in typed context  [no-untyped-call]
src/inmanta/parser/plyInmantaParser.py:0: error: Call to untyped function "Not" in typed context  [no-untyped-call]
src/inmanta/parser/plyInmantaParser.py:0: error: Incompatible types in assignment (expression has type "Literal", variable has type "CreateList")  [assignment]
src/inmanta/parser/plyInmantaParser.py:0: error: Incompatible types in assignment (expression has type "Namespace | None", variable has type "Namespace")  [assignment]
src/inmanta/plugins.py:0: error: "Plugin" has no attribute "__function_name__"  [attr-defined]
src/inmanta/plugins.py:0: error: "Plugin" has no attribute "call"  [attr-defined]
src/inmanta/plugins.py:0: error: "Plugin" has no attribute "call"  [attr-defined]
src/inmanta/plugins.py:0: error: "Plugin" has no attribute "opts"  [attr-defined]
src/inmanta/plugins.py:0: error: "Plugin" has no attribute "opts"  [attr-defined]
src/inmanta/plugins.py:0: error: "Plugin" has no attribute "opts"  [attr-defined]
src/inmanta/plugins.py:0: error: "type[Plugin]" has no attribute "__fq_plugin_name__"  [attr-defined]
src/inmanta/plugins.py:0: error: "type[Plugin]" has no attribute "__function__"  [attr-defined]
src/inmanta/plugins.py:0: error: "type[Plugin]" has no attribute "__function__"  [attr-defined]
src/inmanta/plugins.py:0: error: "type[Plugin]" has no attribute "__function__"  [attr-defined]
src/inmanta/plugins.py:0: error: "type[Plugin]" has no attribute "__function__"  [attr-defined]
src/inmanta/plugins.py:0: error: "type[Plugin]" has no attribute "__function_name__"  [attr-defined]
src/inmanta/plugins.py:0: error: "type[Plugin]" has no attribute "__function_name__"  [attr-defined]
src/inmanta/plugins.py:0: error: "type[Plugin]" has no attribute "__function_name__"  [attr-defined]
src/inmanta/plugins.py:0: error: Argument 1 to "UnConvertibleEntity" has incompatible type "Type"; expected "Entity"  [arg-type]
src/inmanta/plugins.py:0: error: Argument 1 to "add_function" of "PluginMeta" has incompatible type "PluginMeta"; expected "type[Plugin]"  [arg-type]
src/inmanta/plugins.py:0: error: Argument 1 to "to_dsl_type" has incompatible type "object"; expected "type[object]"  [arg-type]
src/inmanta/plugins.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/plugins.py:0: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/inmanta/plugins.py:0: error: Incompatible return value type (got "None", expected "DynamicStatement")  [return-value]
src/inmanta/plugins.py:0: error: Invalid index type "object" for "dict[str | None, Type]"; expected type "str | None"  [index]
src/inmanta/plugins.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/plugins.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/plugins.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/plugins.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
src/inmanta/plugins.py:0: error: Module has no attribute "get_generic_bases"  [attr-defined]
src/inmanta/plugins.py:0: error: Module has no attribute "get_generic_bases"  [attr-defined]
src/inmanta/plugins.py:0: error: Returning Any from function declared to return "Callable[..., Any]"  [no-any-return]
src/inmanta/plugins.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/plugins.py:0: error: Value of type "tuple[Any, ...] | None" is not indexable  [index]
src/inmanta/protocol/auth/decorators.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/protocol/auth/policy_engine.py:0: error: Returning Any from function declared to return "dict[str, object]"  [no-any-return]
src/inmanta/protocol/auth/providers.py:0: error: Item "bool" of "str | bool | Sequence[str] | Mapping[str, str]" has no attribute "__iter__" (not iterable)  [union-attr]
src/inmanta/protocol/common.py:0: error: "Callable[..., BaseModel | UUID | bool | int | float | datetime | str | None | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None | ReturnValue[BaseModel | UUID | bool | int | float | datetime | str | None | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None]]" has no attribute "__method_properties__"  [attr-defined]
src/inmanta/protocol/common.py:0: error: "Callable[..., BaseModel | UUID | bool | int | float | datetime | str | None | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None | ReturnValue[BaseModel | UUID | bool | int | float | datetime | str | None | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None]]" has no attribute "__method_properties__"  [attr-defined]
src/inmanta/protocol/common.py:0: error: "Callable[[Result], dict[str, Any] | None]" has no attribute "setter"  [attr-defined]
src/inmanta/protocol/common.py:0: error: "type" has no attribute "__supertype__"  [attr-defined]
src/inmanta/protocol/common.py:0: error: Argument "text" to "parse" has incompatible type "str | None"; expected "str"  [arg-type]
src/inmanta/protocol/common.py:0: error: Argument 1 to "_get_http_status_code_for_exception" of "MethodProperties" has incompatible type "str | None"; expected "str"  [arg-type]
src/inmanta/protocol/common.py:0: error: Argument 2 to "get_body" of "ReturnValue" has incompatible type "str | None"; expected "str"  [arg-type]
src/inmanta/protocol/common.py:0: error: Dict entry 0 has incompatible type "str": "dict[str, list[str]]"; expected "str": "BaseModel | UUID | int | float | datetime | str | None"  [dict-item]
src/inmanta/protocol/common.py:0: error: Incompatible return value type (got "dict[str, Any | list[Any]]", expected "dict[str, str]")  [return-value]
src/inmanta/protocol/common.py:0: error: Missing type parameters for generic type "ReturnValue"  [type-arg]
src/inmanta/protocol/common.py:0: error: Need type annotation for "params_to_add" (hint: "params_to_add: dict[<type>, <type>] = ...")  [var-annotated]
src/inmanta/protocol/common.py:0: error: No overload variant of "create_model" matches argument types "str", "dict[str, tuple[object, object | None]]", "type[DateTimeNormalizerModel]"  [call-overload]
src/inmanta/protocol/common.py:0: error: Returning Any from function declared to return "int"  [no-any-return]
src/inmanta/protocol/common.py:0: error: Returning Any from function declared to return "type[BaseModel]"  [no-any-return]
src/inmanta/protocol/common.py:0: error: Untyped decorator makes function "set_result" untyped  [misc]
src/inmanta/protocol/common.py:0: note: Possible overload variants:
src/inmanta/protocol/common.py:0: note: def [ModelT: BaseModel] create_model(str, /, *, __config__: ConfigDict | None = ..., __doc__: str | None = ..., __base__: type[ModelT] | tuple[type[ModelT], ...], __module__: str = ..., __validators__: dict[str, Callable[..., Any]] | None = ..., __cls_kwargs__: dict[str, Any] | None = ..., **field_definitions: Any | tuple[str, Any]) -> type[ModelT]
src/inmanta/protocol/common.py:0: note: def create_model(str, /, *, __config__: ConfigDict | None = ..., __doc__: str | None = ..., __base__: None = ..., __module__: str = ..., __validators__: dict[str, Callable[..., Any]] | None = ..., __cls_kwargs__: dict[str, Any] | None = ..., **field_definitions: Any | tuple[str, Any]) -> type[BaseModel]
src/inmanta/protocol/decorators.py:0: error: "FuncT" has no attribute "__api_version__"  [attr-defined]
src/inmanta/protocol/decorators.py:0: error: "FuncT" has no attribute "__protocol_mapping__"  [attr-defined]
src/inmanta/protocol/decorators.py:0: error: "FuncT" has no attribute "__protocol_method__"  [attr-defined]
src/inmanta/protocol/decorators.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/protocol/decorators.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/protocol/endpoints.py:0: error: "Callable[..., Any]" has no attribute "__api_version__"  [attr-defined]
src/inmanta/protocol/endpoints.py:0: error: "Callable[..., Any]" has no attribute "__api_version__"  [attr-defined]
src/inmanta/protocol/endpoints.py:0: error: Argument "args" to "_call" of "Client" has incompatible type "tuple[object, ...]"; expected "list[object]"  [arg-type]
src/inmanta/protocol/endpoints.py:0: error: Argument 1 to "BaseHttpException" has incompatible type "Any | None"; expected "int"  [arg-type]
src/inmanta/protocol/endpoints.py:0: error: Argument 1 to "run_coroutine_threadsafe" has incompatible type "Awaitable[Result]"; expected "Coroutine[Any, Any, Never]"  [arg-type]
src/inmanta/protocol/endpoints.py:0: error: Argument 1 to "update" of "MutableMapping" has incompatible type "dict[str, str] | None"; expected "SupportsKeysAndGetItem[str, Any]"  [arg-type]
src/inmanta/protocol/endpoints.py:0: error: Call to untyped function "close" in typed context  [no-untyped-call]
src/inmanta/protocol/endpoints.py:0: error: Call to untyped function "close" in typed context  [no-untyped-call]
src/inmanta/protocol/endpoints.py:0: error: Cannot determine type of "key"  [has-type]
src/inmanta/protocol/endpoints.py:0: error: Cannot determine type of "key"  [has-type]
src/inmanta/protocol/endpoints.py:0: error: Cannot determine type of "value"  [has-type]
src/inmanta/protocol/endpoints.py:0: error: Cannot determine type of "value"  [has-type]
src/inmanta/protocol/endpoints.py:0: error: Cannot determine type of "value"  [has-type]
src/inmanta/protocol/endpoints.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/endpoints.py:0: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/inmanta/protocol/endpoints.py:0: error: Incompatible types in assignment (expression has type "Any | BaseModel | UUID | int | float | datetime | str | None", variable has type "str")  [assignment]
src/inmanta/protocol/endpoints.py:0: error: Incompatible types in assignment (expression has type "None", variable has type "RESTClient")  [assignment]
src/inmanta/protocol/endpoints.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/protocol/endpoints.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/protocol/endpoints.py:0: error: Need type annotation for "ta"  [var-annotated]
src/inmanta/protocol/endpoints.py:0: error: No overload variant of "__getitem__" of "Sequence" matches argument type "str"  [call-overload]
src/inmanta/protocol/endpoints.py:0: error: Return type "Coroutine[Any, Any, BaseModel | UUID | bool | int | float | datetime | str | None | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None]" of "_call" incompatible with return type "Coroutine[Any, Any, Result]" in supertype "Client"  [override]
src/inmanta/protocol/endpoints.py:0: error: Returning Any from function declared to return "BaseModel | UUID | bool | int | float | datetime | str | None | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None"  [no-any-return]
src/inmanta/protocol/endpoints.py:0: error: Unpacking a string is disallowed  [misc]
src/inmanta/protocol/endpoints.py:0: error: Unsupported right operand type for in ("BaseModel | UUID | int | float | datetime | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None]")  [operator]
src/inmanta/protocol/endpoints.py:0: error: Value of type "BaseModel | UUID | int | float | datetime | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None]" is not indexable  [index]
src/inmanta/protocol/endpoints.py:0: note: Possible overload variants:
src/inmanta/protocol/endpoints.py:0: note: Use "-> None" if function does not return a value
src/inmanta/protocol/endpoints.py:0: note: def __getitem__(self, int, /) -> BaseModel | UUID | bool | int | float | datetime | str | None
src/inmanta/protocol/endpoints.py:0: note: def __getitem__(self, slice[Any, Any, Any], /) -> Sequence[BaseModel | UUID | bool | int | float | datetime | str | None]
src/inmanta/protocol/ipc_light.py:0: error: Argument 1 of "connection_made" is incompatible with supertype "asyncio.protocols.BaseProtocol"; supertype defines the argument type as "BaseTransport"  [override]
src/inmanta/protocol/ipc_light.py:0: error: Invalid index type "UUID | None" for "dict[UUID, Future[object]]"; expected type "UUID"  [index]
src/inmanta/protocol/ipc_light.py:0: error: Item "None" of "Transport | None" has no attribute "close"  [union-attr]
src/inmanta/protocol/ipc_light.py:0: error: Item "None" of "Transport | None" has no attribute "is_closing"  [union-attr]
src/inmanta/protocol/ipc_light.py:0: error: Item "None" of "Transport | None" has no attribute "write"  [union-attr]
src/inmanta/protocol/ipc_light.py:0: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
src/inmanta/protocol/ipc_light.py:0: note: This violates the Liskov substitution principle
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/methods.py:0: error: Incompatible return value type (got "UUID", expected "Environment")  [return-value]
src/inmanta/protocol/methods.py:0: error: Incompatible types in assignment (expression has type "Environment | None", variable has type "UUID")  [assignment]
src/inmanta/protocol/methods.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "list"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "list"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "list"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "list"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "list"  [type-arg]
src/inmanta/protocol/methods.py:0: error: Missing type parameters for generic type "list"  [type-arg]
src/inmanta/protocol/methods.py:0: note: Use "-> None" if function does not return a value
src/inmanta/protocol/methods.py:0: note: Use "-> None" if function does not return a value
src/inmanta/protocol/methods.py:0: note: Use "-> None" if function does not return a value
src/inmanta/protocol/methods_v2.py:0: error: Cannot determine type of "RVID_OPTS"  [has-type]
src/inmanta/protocol/methods_v2.py:0: error: Cannot determine type of "RVID_OPTS"  [has-type]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Missing return statement  [empty-body]
src/inmanta/protocol/methods_v2.py:0: error: Module "inmanta.protocol.methods" does not explicitly export attribute "ArgOption"  [attr-defined]
src/inmanta/protocol/methods_v2.py:0: error: Module "inmanta.protocol.methods" does not explicitly export attribute "ArgOption"  [attr-defined]
src/inmanta/protocol/openapi/converter.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/protocol/openapi/converter.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/protocol/openapi/converter.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/protocol/openapi/converter.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/protocol/openapi/converter.py:0: error: "object" has no attribute "keys"  [attr-defined]
src/inmanta/protocol/openapi/converter.py:0: error: Argument "parameters" to "Operation" has incompatible type "list[Parameter] | None"; expected "list[Parameter | Reference] | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument "url" to "Server" has incompatible type "str"; expected "AnyUrl"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "Schema | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "Schema | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "Schema | bool | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "Schema | bool | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "Sequence[Schema] | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "Sequence[Schema] | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "bool | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "bool | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "dict[str, Schema] | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "dict[str, Schema] | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "list[str] | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "list[str] | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "str | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "Schema" has incompatible type "**dict[str, object]"; expected "str | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "_handle_pydantic_model" of "OpenApiTypeConverter" has incompatible type "type[Sub]"; expected "BaseModel"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "_handle_pydantic_model_recursively" of "OpenApiTypeConverter" has incompatible type "object"; expected "dict[str, object]"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 1 to "len" has incompatible type "object"; expected "Sized"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Argument 7 to "Operation" has incompatible type "**dict[str, RequestBody]"; expected "bool | None"  [arg-type]
src/inmanta/protocol/openapi/converter.py:0: error: Call to untyped function "ssl_enabled" in typed context  [no-untyped-call]
src/inmanta/protocol/openapi/converter.py:0: error: Incompatible return value type (got "list[Parameter | Reference]", expected "list[Parameter]")  [return-value]
src/inmanta/protocol/openapi/converter.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/protocol/openapi/converter.py:0: error: Missing type parameters for generic type "dict"  [type-arg]
src/inmanta/protocol/openapi/converter.py:0: error: Unsupported target for indexed assignment ("object")  [index]
src/inmanta/protocol/openapi/converter.py:0: error: Unsupported target for indexed assignment ("object")  [index]
src/inmanta/protocol/openapi/converter.py:0: error: Value of type "dict[str, Schema] | None" is not indexable  [index]
src/inmanta/protocol/openapi/converter.py:0: error: Value of type "dict[str, Schema] | None" is not indexable  [index]
src/inmanta/protocol/openapi/converter.py:0: error: Value of type "object" is not indexable  [index]
src/inmanta/protocol/openapi/converter.py:0: error: Value of type "object" is not indexable  [index]
src/inmanta/protocol/openapi/converter.py:0: error: Variable "type_annotation" is not valid as a type  [valid-type]
src/inmanta/protocol/openapi/converter.py:0: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/inmanta/protocol/rest/__init__.py:0: error: Argument "status_code" to "ReturnValue" has incompatible type "int | BaseModel | UUID | float | datetime | str | None"; expected "int"  [arg-type]
src/inmanta/protocol/rest/__init__.py:0: error: Argument "status_code" to "ReturnValue" has incompatible type "int | BaseModel | UUID | float | datetime | str | None"; expected "int"  [arg-type]
src/inmanta/protocol/rest/__init__.py:0: error: Argument 1 to "get" of "Mapping" has incompatible type "str | None"; expected "str"  [arg-type]
src/inmanta/protocol/rest/__init__.py:0: error: Argument 2 has incompatible type "dict[str, object]"; expected "dict[str, str]"  [arg-type]
src/inmanta/protocol/rest/__init__.py:0: error: Call to untyped function "_validate_argument_consistency" in typed context  [no-untyped-call]
src/inmanta/protocol/rest/__init__.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/protocol/rest/__init__.py:0: error: Incompatible types in assignment (expression has type "object | None", variable has type "str | None")  [assignment]
src/inmanta/protocol/rest/__init__.py:0: error: Item "bool" of "str | bool | Sequence[str] | Mapping[str, str]" has no attribute "__iter__" (not iterable)  [union-attr]
src/inmanta/protocol/rest/__init__.py:0: error: Returning Any from function declared to return "object | None"  [no-any-return]
src/inmanta/protocol/rest/client.py:0: error: Argument "body" to "HTTPRequest" has incompatible type "dict[str, Any] | None"; expected "bytes | str | None"  [arg-type]
src/inmanta/protocol/rest/client.py:0: error: Argument "result" to "Result" has incompatible type "bytes"; expected "dict[str, Any] | None"  [arg-type]
src/inmanta/protocol/rest/client.py:0: error: Argument "result" to "Result" has incompatible type "str"; expected "dict[str, Any] | None"  [arg-type]
src/inmanta/protocol/rest/client.py:0: error: Argument "result" to "Result" has incompatible type "str"; expected "dict[str, Any] | None"  [arg-type]
src/inmanta/protocol/rest/client.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/protocol/rest/client.py:0: error: Incompatible types in assignment (expression has type "bytes | str", variable has type "dict[str, Any] | None")  [assignment]
src/inmanta/protocol/rest/client.py:0: error: Incompatible types in assignment (expression has type "str | None", variable has type "str")  [assignment]
src/inmanta/protocol/rest/client.py:0: error: Incompatible types in assignment (expression has type "str | int | None", variable has type "int")  [assignment]
src/inmanta/protocol/rest/client.py:0: error: Incompatible types in assignment (expression has type "str | int | None", variable has type "int")  [assignment]
src/inmanta/protocol/rest/client.py:0: error: Value expression in dictionary comprehension has incompatible type "str"; expected type "bytes"  [misc]
src/inmanta/protocol/rest/client.py:0: note: Use "-> None" if function does not return a value
src/inmanta/protocol/rest/server.py:0: error: Argument 1 to "get_global_url_map" of "RESTServer" has incompatible type "Sequence[CallTarget]"; expected "list[CallTarget]"  [arg-type]
src/inmanta/references.py:0: error: Argument "value" to "MutatedJsonArgument" has incompatible type "object"; expected "StrictJson"  [arg-type]
src/inmanta/references.py:0: error: Argument 1 to "update" of "JSONPath" has incompatible type "Resource"; expected "JsonValue"  [arg-type]
src/inmanta/references.py:0: error: Incompatible types in assignment (expression has type "dict[str, StrictJson] | list[StrictJson] | str | int | float | None", variable has type "dict[str, Any]")  [assignment]
src/inmanta/references.py:0: error: Parameter 1 of Literal[...] is invalid  [valid-type]
src/inmanta/references.py:0: error: Returning Any from function declared to return "type[T] | None"  [no-any-return]
src/inmanta/resources.py:0: error: "Resource" has no attribute "receive_events"; maybe "get_receive_events"?  [attr-defined]
src/inmanta/resources.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/resources.py:0: error: Incompatible return value type (got "Resource", expected "R")  [return-value]
src/inmanta/resources.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/resources.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/server/agentmanager.py:0: error: Item "None" of "Sequence[Agent] | None" has no attribute "__iter__" (not iterable)  [union-attr]
src/inmanta/server/agentmanager.py:0: error: Value of type "dict[str, Any] | None" is not indexable  [index]
src/inmanta/server/agentmanager.py:0: error: Value of type variable "AnyStr" of "abspath" cannot be "str | None"  [type-var]
src/inmanta/server/bootloader.py:0: error: Incompatible return value type (got "Iterator[ModuleInfo]", expected "Generator[ModuleInfo, None, None]")  [return-value]
src/inmanta/server/config.py:0: error: Cannot infer type argument 1 of "Option"  [misc]
src/inmanta/server/config.py:0: error: Cannot infer type argument 1 of "Option"  [misc]
src/inmanta/server/config.py:0: error: Cannot infer type argument 1 of "Option"  [misc]
src/inmanta/server/config.py:0: error: Cannot infer type argument 1 of "Option"  [misc]
src/inmanta/server/config.py:0: error: Cannot infer type argument 1 of "Option"  [misc]
src/inmanta/server/config.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/server/config.py:0: error: No overload variant of "int" matches argument type "object"  [call-overload]
src/inmanta/server/config.py:0: error: Returning Any from function declared to return "int"  [no-any-return]
src/inmanta/server/config.py:0: note: Possible overload variants:
src/inmanta/server/config.py:0: note: def __new__(cls, str | Buffer | SupportsInt | SupportsIndex | SupportsTrunc = ..., /) -> int
src/inmanta/server/config.py:0: note: def __new__(cls, str | bytes | bytearray, /, base: SupportsIndex) -> int
src/inmanta/server/diff.py:0: error: "object" has no attribute "__iter__"; maybe "__dir__" or "__str__"? (not iterable)  [attr-defined]
src/inmanta/server/diff.py:0: error: Incompatible return value type (got "str | None", expected "str")  [return-value]
src/inmanta/server/extensions.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/server/protocol.py:0: error: Argument "result" to "Result" has incompatible type "BaseModel | UUID | int | float | datetime | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None"; expected "dict[str, Any] | None"  [arg-type]
src/inmanta/server/protocol.py:0: error: Argument 1 to "RESTServer" has incompatible type "SessionManager"; expected "SessionManagerInterface"  [arg-type]
src/inmanta/server/protocol.py:0: error: Argument 1 to "append" of "list" has incompatible type "tuple[str, type[RedirectHandler], dict[str, str]]"; expected "Rule"  [arg-type]
src/inmanta/server/protocol.py:0: error: Argument 1 to "start" of "RESTServer" has incompatible type "dict_values[str, ServerSlice]"; expected "Sequence[CallTarget]"  [arg-type]
src/inmanta/server/protocol.py:0: error: Argument 2 to "_execute_call" of "RESTBase" has incompatible type "dict[str, Any] | None"; expected "dict[str, object]"  [arg-type]
src/inmanta/server/protocol.py:0: error: Dict entry 0 has incompatible type "str": "list[Request]"; expected "str": "str"  [dict-item]
src/inmanta/server/protocol.py:0: error: Incompatible return value type (got "BaseModel | UUID | int | float | datetime | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None", expected "Result")  [return-value]
src/inmanta/server/protocol.py:0: error: Item "None" of "HTTPServer | None" has no attribute "_sockets"  [union-attr]
src/inmanta/server/protocol.py:0: error: Item "None" of "HTTPServer | None" has no attribute "_sockets"  [union-attr]
src/inmanta/server/protocol.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/server/protocol.py:0: error: Missing type parameters for generic type "Future"  [type-arg]
src/inmanta/server/protocol.py:0: error: Missing type parameters for generic type "Future"  [type-arg]
src/inmanta/server/protocol.py:0: error: Missing type parameters for generic type "Future"  [type-arg]
src/inmanta/server/protocol.py:0: error: Missing type parameters for generic type "Future"  [type-arg]
src/inmanta/server/protocol.py:0: error: Module "tornado.gen" does not explicitly export attribute "TimeoutError"  [attr-defined]
src/inmanta/server/protocol.py:0: error: Need type annotation for "future"  [var-annotated]
src/inmanta/server/protocol.py:0: error: Return type "Coroutine[Any, Any, Result]" of "_call" incompatible with return type "Coroutine[Any, Any, BaseModel | UUID | bool | int | float | datetime | str | None | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None] | None]" in supertype "inmanta.protocol.endpoints.TypedClient"  [override]
src/inmanta/server/server.py:0: error: Argument 1 to "get_global_url_map" of "RESTServer" has incompatible type "dict_values[str, ServerSlice]"; expected "list[CallTarget]"  [arg-type]
src/inmanta/server/services/environment_metrics_service.py:0: error: "Sequence[EnvironmentMetricsGauge]" has no attribute "append"  [attr-defined]
src/inmanta/server/services/environment_metrics_service.py:0: error: "Sequence[EnvironmentMetricsTimer]" has no attribute "append"  [attr-defined]
src/inmanta/server/services/environment_metrics_service.py:0: error: "str" has no attribute "value"  [attr-defined]
src/inmanta/server/services/environment_metrics_service.py:0: error: Argument 1 to "create_metrics_timer" has incompatible type "Sequence[MetricValue]"; expected "Sequence[MetricValueTimer]"  [arg-type]
src/inmanta/server/services/environment_metrics_service.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/server/services/environment_metrics_service.py:0: error: Function is missing a return type annotation  [no-untyped-def]
src/inmanta/server/services/environment_metrics_service.py:0: error: Unsupported target for indexed assignment ("float | dict[str, float] | None")  [index]
src/inmanta/server/services/metricservice.py:0: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
src/inmanta/server/services/orchestrationservice.py:0: error: Argument 1 to "add_background_task" of "TaskHandler" has incompatible type "Coroutine[Any, Any, tuple[int, dict[str, Any] | None]]"; expected "Coroutine[object, None, Result | None]"  [arg-type]
src/inmanta/server/services/orchestrationservice.py:0: error: Argument 1 to "append" of "list" has incompatible type "Resource"; expected "dict[str, Any]"  [arg-type]
src/inmanta/server/services/orchestrationservice.py:0: error: Incompatible return value type (got "ReturnValueWithMeta[Sequence[DesiredStateVersion]]", expected "ReturnValue[list[DesiredStateVersion]]")  [return-value]
src/inmanta/server/services/orchestrationservice.py:0: error: Incompatible types in assignment (expression has type "dict[Never, Never]", target has type "UUID | bool | int | float | datetime | str | None")  [assignment]
src/inmanta/server/services/orchestrationservice.py:0: error: Incompatible types in assignment (expression has type "set[str | None]", variable has type "set[str]")  [assignment]
src/inmanta/server/services/orchestrationservice.py:0: error: Unsupported target for indexed assignment ("Resource")  [index]
src/inmanta/server/services/orchestrationservice.py:0: error: Value of type "Resource" is not indexable  [index]
src/inmanta/server/services/paramservice.py:0: error: Unsupported operand types for + ("None" and "timedelta")  [operator]
src/inmanta/server/services/paramservice.py:0: error: Unsupported operand types for > ("timedelta" and "datetime")  [operator]
src/inmanta/server/services/paramservice.py:0: note: Left operand is of type "datetime | None"
src/inmanta/server/services/paramservice.py:0: note: Left operand is of type "datetime | timedelta"
src/inmanta/server/services/resourceservice.py:0: error: "ServerSlice" has no attribute "register_listener_for_multiple_actions"  [attr-defined]
src/inmanta/server/services/resourceservice.py:0: error: Argument "args" to "__init__" of "LogRecord" has incompatible type "list[Never]"; expected "tuple[object, ...] | Mapping[str, object] | None"  [arg-type]
src/inmanta/server/services/resourceservice.py:0: error: Missing type parameters for generic type "Feature"  [type-arg]
src/inmanta/server/services/resourceservice.py:0: error: Module "inmanta.server.protocol" does not explicitly export attribute "handle"  [attr-defined]
src/inmanta/server/services/resourceservice.py:0: error: Module has no attribute "_startTime"  [attr-defined]
src/inmanta/server/services/resourceservice.py:0: error: Unsupported target for indexed assignment ("Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None | Sequence[BaseModel | UUID | bool | int | float | datetime | str | None] | Mapping[str, BaseModel | UUID | bool | int | float | datetime | str | None]] | None")  [index]
src/inmanta/server/services/resourceservice.py:0: error: Value of type variable "FuncT" of "__call__" of "handle" cannot be "Callable[[ResourceService, Environment, Id, Change | None], Coroutine[Any, Any, dict[ResourceIdStr, list[ResourceAction]]]]"  [type-var]
src/inmanta/server/services/userservice.py:0: error: Module "inmanta.data" does not explicitly export attribute "AuthMethod"  [attr-defined]
src/inmanta/server/validate_filter.py:0: error: Incompatible return value type (got "Callable[[object], list[tuple[RangeOperator, datetime]] | None]", expected "Callable[[object, object], list[tuple[RangeOperator, datetime]] | None]")  [return-value]
src/inmanta/server/validate_filter.py:0: error: Incompatible return value type (got "Callable[[object], list[tuple[RangeOperator, int]] | None]", expected "Callable[[object, object], list[tuple[RangeOperator, int]] | None]")  [return-value]
src/inmanta/server/validate_filter.py:0: error: Incompatible return value type (got "dict[QueryType, list[Any]] | None", expected "dict[QueryType, ReleasedResourceState] | None")  [return-value]
src/inmanta/server/validate_filter.py:0: error: Too few arguments  [call-arg]
src/inmanta/server/validate_filter.py:0: error: Too few arguments  [call-arg]
src/inmanta/server/validate_filter.py:0: error: Unexpected keyword argument "field" for "Filter"  [call-arg]
src/inmanta/signals.py:0: error: Argument 2 to "signal" has incompatible type "Callable[[Signals, FrameType | None], None]"; expected "Callable[[int, FrameType | None], Any] | int | Handlers | None"  [arg-type]
src/inmanta/signals.py:0: error: Argument 2 to "signal" has incompatible type "Callable[[Signals, FrameType | None], None]"; expected "Callable[[int, FrameType | None], Any] | int | Handlers | None"  [arg-type]
src/inmanta/signals.py:0: error: Argument 2 to "signal" has incompatible type "Callable[[Signals, FrameType | None], None]"; expected "Callable[[int, FrameType | None], Any] | int | Handlers | None"  [arg-type]
src/inmanta/signals.py:0: error: Skipping analyzing "rpdb": module is installed, but missing library stubs or py.typed marker  [import-untyped]
src/inmanta/stable_api.py:0: error: Missing type parameters for generic type "Callable"  [type-arg]
src/inmanta/tracing.py:0: error: Call to untyped function "AsyncPGInstrumentor" in typed context  [no-untyped-call]
src/inmanta/types.py:0: error: "type" has no attribute "__supertype__"  [attr-defined]
src/inmanta/util/__init__.py:0: error: Argument "key" to "sorted" has incompatible type "Callable[[T], S]"; expected "Callable[[T], SupportsDunderLT[Any] | SupportsDunderGT[Any]]"  [arg-type]
src/inmanta/util/__init__.py:0: error: Incompatible return value type (got "bool | Awaitable[bool]", expected "bool")  [return-value]
src/inmanta/util/__init__.py:0: error: Returning Any from function declared to return "bool"  [no-any-return]
src/inmanta/util/async_lru.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/util/async_lru.py:0: error: Function is missing a type annotation  [no-untyped-def]
src/inmanta/util/async_lru.py:0: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/inmanta/validation_type.py:0: error: Argument 1 to "parametrize_type" has incompatible type "object"; expected "type[object] | Callable[..., type[object]]"  [arg-type]
src/inmanta/validation_type.py:0: error: Incompatible types in assignment (expression has type "object", variable has type "Mapping[str, object] | None")  [assignment]
src/inmanta/vendor/pyformance/registry.py:0: error: Incompatible types in assignment (expression has type "Histogram | Meter | Gauge[int | float] | Timer | Counter", target has type "Counter")  [assignment]
src/inmanta/vendor/pyformance/registry.py:0: error: Incompatible types in assignment (expression has type "Histogram | Meter | Gauge[int | float] | Timer | Counter", target has type "Gauge[int | float]")  [assignment]
src/inmanta/vendor/pyformance/registry.py:0: error: Incompatible types in assignment (expression has type "Histogram | Meter | Gauge[int | float] | Timer | Counter", target has type "Histogram")  [assignment]
src/inmanta/vendor/pyformance/registry.py:0: error: Incompatible types in assignment (expression has type "Histogram | Meter | Gauge[int | float] | Timer | Counter", target has type "Meter")  [assignment]
src/inmanta/vendor/pyformance/registry.py:0: error: Incompatible types in assignment (expression has type "Histogram | Meter | Gauge[int | float] | Timer | Counter", target has type "Timer")  [assignment]
src/inmanta/warnings.py:0: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/inmanta/warnings.py:0: error: Name "logger" already defined on line 0  [no-redef]
src/inmanta/warnings.py:0: error: Name "text" already defined on line 0  [no-redef]
src/inmanta/warnings.py:0: error: Unused "type: ignore" comment  [unused-ignore]
